### PR TITLE
[now-cli] Add support for new keys in `now dev`

### PR DESCRIPTION
**Pull request by styfle**
This PR is a followup to #3138 so that `now dev` will validate and transform the following `now.json` config keys to routes:

- cleanUrls
- rewrites
- redirects
- headers
- trailingSlash

[PRODUCT-341] #close

[PRODUCT-341]: https://zeit.atlassian.net/browse/PRODUCT-341

### CODE REVIEW

File: packages/now-cli/src/util/dev/router.ts [hunk 1]
```diff
@@ -40,28 +41,21 @@ export default async function(
...
         continue;
       }
 
+      let { src, headers, methods } = routeConfig;
+
       if (Array.isArray(methods) && reqMethod && !methods.includes(reqMethod)) {
         continue;
       }
 
-      if (!src.startsWith('^')) {
```
Comment (TooTallNate) [inline packages/now-cli/src/util/dev/router.ts:57]: Why are these removed?

Comment (styfle) [reply packages/now-cli/src/util/dev/router.ts:57]: Because its handled in `@now/routing-utils`

Comment (styfle) [reply packages/now-cli/src/util/dev/router.ts:57]: Take a look here: https://github.com/zeit/now/blob/110e1445fa267105fea86554f1d4b2825ff07791/packages/now-routing-utils/src/index.ts#L59

**Review by TooTallNate — APPROVED** [review]
